libraries/AP_HAL/UARTDriver.cpp

```diff
@@ -0,0 +1,49 @@
+#include "UARTDriver.h"
+
+#if HAL_LOG_UART_STATS
+
+#include <AP_Logger/AP_Logger.h>
+
+extern const AP_HAL::HAL &hal;
+
+void AP_HAL::UARTDriver::update_logging_all()
+{
+    hal.uartA->update_logging();
```

Comment:
It would be a lot nicer if we did...

Found in AP_SerialManager for the init:
``` c++
    // initialise serial ports    
    for (uint8_t i=1; i<SERIALMANAGER_NUM_PORTS; i++) {    
        auto *uart = hal.serial(i);  
```

Reply:
> It would be a lot nicer if we did...
> 
> Found in AP_SerialManager for the init:
> 
> ```c++
>     // initialise serial ports    
>     for (uint8_t i=1; i<SERIALMANAGER_NUM_PORTS; i++) {    
>         auto *uart = hal.serial(i);  
> ```

Thanks.

Interestingly different from other places using `num_serial` in place of `SERIALMANAGER_NUM_PORTS`